chains/substrate/listener/event-handlers.go

```diff
@@ -161,14 +181,16 @@ func (rh *RetryEventHandler) HandleEvents(evts []*parser.Event, msgChan chan []*
 						var d events.Deposit
 						err = mapstructure.Decode(event.Fields, &d)
 						if err != nil {
+							span.SetStatus(codes.Error, err.Error())
 							return err
 						}
 						m, err := rh.depositHandler.HandleDeposit(rh.domainID, d.DestDomainID, d.DepositNonce, d.ResourceID, d.CallData, d.TransferType)
 						if err != nil {
+							span.SetStatus(codes.Error, err.Error())
```

Comment:
I'm not sure if these actually give any usefull information or the same effect we would have if you just set parent span (listener one) to error if the error is returned because we would see it errored on the last event handler called and we wouldn't need to set span to error for each return which is kinda bloatey.

Reply:
Would do that approach everywhere we have the same logic.